From 82ee83a8e1ab706eadf316903a83328fc853da32 Mon Sep 17 00:00:00 2001
From: Dinh Nguyen <dinguyen@altera.com>
Date: Thu, 22 May 2014 09:55:13 -0500
Subject: [PATCH 002/172] FogBugz #198256: Fix unnecessary USB overcurrent
 condition

The CycloneV and ArriaV devkit has a small C_VBUS capacitor on the devkit.
Because of this, the USB driver sometimes detects an unnecessary
overcurrent error condition.

This patch uses the external VbusValid signal instead.

Signed-off-by: Dinh Nguyen <dinguyen@altera.com>
[czou:Original patch taken from
https://github.com/altera-opensource/linux-socfpga.git socfpga-4.1]
Signed-off-by: czou <cao.zou@windriver.com>
---
 drivers/usb/dwc2/core.c | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/drivers/usb/dwc2/core.c b/drivers/usb/dwc2/core.c
index d5197d4..ccc9b6c 100644
--- a/drivers/usb/dwc2/core.c
+++ b/drivers/usb/dwc2/core.c
@@ -422,6 +422,11 @@ int dwc2_core_init(struct dwc2_hsotg *hsotg, bool select_phy, int irq)
 	if (hsotg->core_params->ts_dline > 0)
 		usbcfg |= GUSBCFG_TERMSELDLPULSE;
 
+	/* Set external VBUS indicator as needed. */
+	if (hsotg->core_params->phy_type == DWC2_PHY_TYPE_PARAM_ULPI)
+		usbcfg |= (GUSBCFG_ULPI_INT_VBUS_IND |
+				GUSBCFG_INDICATORPASSTHROUGH);
+
 	writel(usbcfg, hsotg->regs + GUSBCFG);
 
 	/* Reset the Controller */
-- 
1.9.1

